File: frontend/react-app/src/services/CommentService.js
import axios from './axios';

export default class CommentService {

    getAllComments() {
        return axios.get('/comments');
    }

    getCommentsByUserId(userId) {
        return axios.get(`/comments/userId/${userId}`);
    }

    getCommentsByPostId(postId) {
        return axios.get(`/comments/postId/${postId}`);
    }

    getCommentsByUserIdAndPostId(userId, postId) {
        return axios.get(`/comments/userId/${userId}/postId/${postId}`);
    }

    getOneCommentById(commentId) {
        return axios.get(`/comments/${commentId}`);
    }

    createOneComment(commentData) {
        return axios.post('/comments', commentData);
    }

    updateOneCommentById(commentId, commentData) {
        return axios.put(`/comments/${commentId}`, commentData);
    }

    deleteOneCommentById(commentId) {
        return axios.delete(`/comments/${commentId}`);
    }
}

File: backend/demo/src/main/java/com/example/demo/services/CommentService.java
package com.example.demo.services;

import com.example.demo.dtos.requests.CommentCreateRequest;
import com.example.demo.dtos.requests.CommentUpdateRequest;
import com.example.demo.entities.Comment;
import com.example.demo.entities.Post;
import com.example.demo.entities.User;
import com.example.demo.repos.CommentRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class CommentService {

    private final CommentRepository commentRepository;
    private UserService userService;
    private PostService postService;
    public CommentService(CommentRepository commentRepository, UserService userService, PostService postService) {
        this.commentRepository = commentRepository;
        this.userService = userService;
        this.postService = postService;
    }

    public List<Comment> getAllComments() {
        return commentRepository.findAll();
    }

    public List<Comment> getAllCommentsByUserId(Long userId) {
        return commentRepository.findByUserId(userId);
    }

    public List<Comment> getAllCommentsByPostId(Long postId) {
        return commentRepository.findByPostId(postId);
    }

    public List<Comment> getAllCommentsByUserAndPost(Long userId, Long postId) {
        return commentRepository.findByUserIdAndPostId(userId, postId);
    }

    public Comment getOneCommentById(Long commentId) {
        return commentRepository.findById(commentId).orElse(null);
    }

    public Comment createOneComment(CommentCreateRequest commentCreateRequest) {
        User user = userService.getOneUserById(commentCreateRequest.getUserId());
        Post post = postService.getOnePostById(commentCreateRequest.getPostId());
        if(user != null && post!= null){
            Comment newComment = new Comment();
            newComment.setText(commentCreateRequest.getText());
            newComment.setUser(user);
            newComment.setPost(post);
            return commentRepository.save(newComment);
        }else
            return null;
    }

    public Comment updateOneCommentById(Long commentId, CommentUpdateRequest commentUpdateRequest) {
        Optional<Comment> comment = commentRepository.findById(commentId);
        if(comment.isPresent()){
            Comment updateComment = comment.get();
            updateComment.setText(commentUpdateRequest.getText());
            commentRepository.save(updateComment);
            return updateComment;
        }else
            return null;
    }

    public void deleteOneCommentById(Long commentId) {
        commentRepository.deleteById(commentId);
    }
}
